Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,784 @@
+        }
+
+        getSpriteNameAction = other.getSpriteNameAction;
+
+        if (other.funcPositionValidation != null)
+        {
+            funcPositionValidation = (Func<Tile, bool>)other.funcPositionValidation.Clone();
+        }
+
+        tileTypeBuildPermissions = other.tileTypeBuildPermissions;
+
+        LocalizationCode = other.LocalizationCode;
+        UnlocalizedDescription = other.UnlocalizedDescription;
+    }
+
+    /// <summary>
+    /// This event will trigger when the utility has been changed.
+    /// This is means that any change (parameters, job state etc) to the utility will trigger this.
+    /// </summary>
+    public event Action<Utility> Changed;
+
+    /// <summary>
+    /// This event will trigger when the utility has been removed.
+    /// </summary>
+    public event Action<Utility> Removed;
+
+    /// <summary>
+    /// Gets the width of the utility.
+    /// </summary>
+    public int Width 
+    { 
+        get { return 1; }
+    }
+
+    /// <summary>
+    /// Gets the height of the utility.
+    /// </summary>
+    public int Height 
+    { 
+        get { return 1; }
+    }
+
+    /// <summary>
+    /// Gets the tint used to change the color of the utility.
+    /// </summary>
+    /// <value>The Color of the utility.</value>
+    public Color Tint { get; private set; }
+
+    /// <summary>
+    /// Gets the spot where the Character will stand when he is using the utility. This is relative to the bottom
+    /// left tile of the sprite. This can be outside of the actual utility.
+    /// </summary>
+    /// <value>The spot where the Character will stand when he uses the utility.</value>
+    public Vector2 JobSpotOffset { get; private set; }
+
+    /// <summary>
+    /// Gets the EventAction for the current utility.
+    /// These actions are called when an event is called. They get passed the utility
+    /// they belong to, plus a deltaTime (which defaults to 0).
+    /// </summary>
+    /// <value>The event actions that is called on update.</value>
+    public EventActions EventActions { get; private set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the utility is selected by the player or not.
+    /// </summary>
+    /// <value>Whether the utility is selected or not.</value>
+    public bool IsSelected { get; set; }
+
+    /// <summary>
+    /// Gets the BASE tile of the utility. (Large objects can span over multiple tiles).
+    /// This should be RENAMED (possibly to BaseTile).
+    /// </summary>
+    /// <value>The BASE tile of the utility.</value>
+    public Tile Tile { get; private set; }
+
+    /// <summary>
+    /// Gets the string that defines the type of object the utility is. This gets queried by the visual system to 
+    /// know what sprite to render for this utility.
+    /// </summary>
+    /// <value>The type of the utility.</value>
+    public string ObjectType { get; private set; }
```

Comment:
Could we change this to just Type? We started changing this in other classes. I want to simplify the PrototypeManager, and have Type for all the classes that use will be great.